packages/frontend/admin/src/modules/setup/form.tsx

```diff
@@ -0,0 +1,238 @@
+import { toast } from 'sonner';
+
+import { CreateAdmin } from './create-admin';
+
+export enum CarouselSteps {
+  Welcome = 0,
+  CreateAdmin,
+  SettingsDone,
+}
+
+const Welcome = () => {
+  return (
+    <div
+      className="flex flex-col h-full w-full mt-60 max-lg:items-center max-lg:mt-16"
+      style={{ minHeight: '300px' }}
+    >
+      <h1 className="text-5xl font-extrabold max-lg:text-3xl max-lg:font-bold">
+        Welcome to AFFiNE
+      </h1>
+      <p className="mt-5 font-semibold text-xl max-lg:px-4 max-lg:text-lg">
+        Configure your Self Host AFFiNE with a few simple settings.
+      </p>
+    </div>
+  );
+};
+
+const SettingsDone = () => {
+  return (
+    <div
+      className="flex flex-col h-full w-full mt-60 max-lg:items-center max-lg:mt-16"
+      style={{ minHeight: '300px' }}
+    >
+      <h1 className="text-5xl font-extrabold max-lg:text-3xl max-lg:font-bold">
+        All Settings Done
+      </h1>
+      <p className="mt-5 font-semibold text-xl max-lg:px-4 max-lg:text-lg">
+        AFFiNE is ready to use.
+      </p>
+    </div>
+  );
+};
+
+const CarouselItemElements = {
+  [CarouselSteps.Welcome]: Welcome,
+  [CarouselSteps.CreateAdmin]: CreateAdmin,
+  [CarouselSteps.SettingsDone]: SettingsDone,
+};
+
+export const Form = () => {
+  const [api, setApi] = useState<CarouselApi>();
+  const [current, setCurrent] = useState(0);
+  const [count, setCount] = useState(0);
+
+  const navigate = useNavigate();
+
+  const [nameValue, setNameValue] = useState('');
+  const [emailValue, setEmailValue] = useState('');
+  const [passwordValue, setPasswordValue] = useState('');
+  const [invalidEmail, setInvalidEmail] = useState(false);
+  const [invalidPassword, setInvalidPassword] = useState(false);
+
+  const { data } = useQuery({
+    query: serverConfigQuery,
+  });
+  const passwordLimits = data.serverConfig.credentialsRequirement.password;
+
+  const isCreateAdminStep = current - 1 === CarouselSteps.CreateAdmin;
+
+  const disableContinue =
+    (!nameValue || !emailValue || !passwordValue) && isCreateAdminStep;
+
+  useEffect(() => {
+    if (!api) {
+      return;
+    }
+
+    setCount(api.scrollSnapList().length);
+    setCurrent(api.selectedScrollSnap() + 1);
+
+    api.on('select', () => {
+      setCurrent(api.selectedScrollSnap() + 1);
+    });
+  }, [api]);
+
+  const createAdmin = useCallback(async () => {
+    if (invalidEmail || invalidPassword) return;
+
+    try {
+      const createResponse = await fetch('/api/setup/create-admin-user', {
+        method: 'POST',
+        body: JSON.stringify({
+          email: emailValue,
+          password: passwordValue,
+        }),
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      });
+
+      if (!createResponse.ok) {
+        const errorData = await createResponse.json();
+        throw new Error(errorData.message || 'Failed to create admin');
+      }
+
+      const signInResponse = await fetch('/api/auth/sign-in', {
```

Comment:
user will be automatically signed-in after create-admin-user called

Reply:
This is for user convenience but I will remove it.